code/controllers/subsystem/dcs.dm

```diff
@@ -33,22 +33,28 @@ PROCESSING_SUBSYSTEM_DEF(dcs)
 	var/datum/element/eletype = arguments[1]
 	var/list/fullid = list("[eletype]")
 	var/list/named_arguments = list()
+
 	for(var/i in initial(eletype.id_arg_index) to length(arguments))
 		var/key = arguments[i]
-		var/value
+
 		if(istext(key))
-			value = arguments[key]
-		if(!(istext(key) || isnum(key)))
-			key = REF(key)
-		key = "[key]" // Key is stringified so numbers dont break things
-		if(!isnull(value))
-			if(!(istext(value) || isnum(value)))
-				value = REF(value)
-			named_arguments["[key]"] = value
-		else
+			var/value = arguments[key]
+			if (isnull(value))
+				if (!istext(value) && !isnum(value))
+					value = REF(value)
+				named_arguments[key] = value
+			else
+				fullid += key
+
+			continue
+
+		if (isnum(key))
 			fullid += "[key]"
+		else
+			fullid += REF(key)
```

Comment:
I wonder if this means we run REF for paths?

Reply:
yeah, idk what it returns though

Reply:
~~likely the same thing as running REF for strings~~